src/System.Security.Permissions/src/System/Security/Permissions/PrincipalPermission.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace System.Security.Permissions
+{
+    public sealed partial class PrincipalPermission : System.Security.IPermission, System.Security.ISecurityEncodable, System.Security.Permissions.IUnrestrictedPermission
+    {
+        public PrincipalPermission(System.Security.Permissions.PermissionState state) { }
+        public PrincipalPermission(string name, string role) { }
+        public PrincipalPermission(string name, string role, bool isAuthenticated) { }
+        public System.Security.IPermission Copy() { return default(System.Security.IPermission); }
+        public void Demand() { }
```

Comment:
> Though I guess that one is really a mis-named

Yes.  Regardless of the name, WindowsIdentity.GetCurrent() is how you get information about the current OS-level user.  When the type was created, the only relevant OS was Windows... that's no longer the case.  We could invent new API in addition, but I would like to see us at least make basic functionality like GetCurrent(), Name, etc. work.  Whether we then make impersonation sort of work is another consideration up for debate.